.github/workflows/pull-with-lifecycle-manager.yaml

```diff
@@ -21,6 +21,18 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Cache Go dependencies
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: "./go.mod"
+
+      - name: Cache Binaries
+        id: cache-binaries
+        uses: actions/cache@v3
+        with:
+          path: bin
+          key: ${{ runner.os }}-bin
```

Comment:
it seems the `key` is always same although binary is built newly. I've read the GH action documentation that the binary is only built only if key is different than previous key. However, here `${{ runner.os }}-bin` is the same  for the same OS. 
Can you clarify this?